### PR TITLE
Merge dotnet/Installer `src` and `test` folder histories

### PR DESCRIPTION
**Pull request by MiYanni**
Related: https://github.com/dotnet/installer/pull/17959
Related: https://github.com/dotnet/sdk/pull/38804

## Summary
In an effort to merge the https://github.com/dotnet/installer repo into this one, we decided to keep the history of the `src` and `test` folders. This means this PR contains the contents of those folders from the dotnet/Installer repo, but does not actually utilize those files/projects. The [Phase 1 PR](https://github.com/dotnet/sdk/pull/38804) I have will be updated to include these files as they are from this history, and then modify them to function in the repo. Therefore, the goal of this PR isn't to make anything function, but instead simply provide the history of these files.

If the files impact the build, I'll modify this PR to decouple them from the SDK build process.

### Public
- ~~https://dev.azure.com/dnceng-public/public/_build/results?buildId=636879~~
- https://dev.azure.com/dnceng-public/public/_build/results?buildId=638697

### Internal
- ~~https://dnceng.visualstudio.com/internal/_build/results?buildId=2426433~~
- https://dnceng.visualstudio.com/internal/_build/results?buildId=2427842

### CODE REVIEW

File: test/dotnet-test.Tests/CollectCodeCoverage.runsettings [hunk 1]
```diff
@@ -1,25 +0,0 @@
-<RunSettings>
```
Comment (nagilson) [inline test/dotnet-test.Tests/CollectCodeCoverage.runsettings:1]: Do we want to delete this?

**Review by nagilson — APPROVED** [review]
Overall LGTM, thank you :) I can confirm the file history seems to be preserved.

File: test/dotnet-test.Tests/CollectCodeCoverage.runsettings (hunk 1 above)
Comment (MiYanni) [reply test/dotnet-test.Tests/CollectCodeCoverage.runsettings:1]: I don't even understand how this is deleted. Like, fundamentally. I made sure to select files from this directory to use the SDK's history. So this shouldn't have happened. Even if I undelete it now, I can't preserve the history of this file. I'd have to redo the entire PR for just this file.

Looks like this file was deleted from the Installer repo in 2018. I think the only reasonable solution is to just add it back in this PR and have the history for it be broken.

Comment (MiYanni) [reply test/dotnet-test.Tests/CollectCodeCoverage.runsettings:1]: Oh oh oh. I think I knew what happened. Wait. Let me check something...

File: test/Microsoft.NET.TestFramework/Assertions/CommandResultExtensions.cs [hunk 2]
```diff
@@ -7,9 +7,6 @@ namespace Microsoft.NET.TestFramework.Assertions
 {
     public static class CommandResultExtensions
     {
-        public static CommandResultAssertions Should(this CommandResult commandResult)
-        {
-            return new CommandResultAssertions(commandResult);
-        }
+        public static CommandResultAssertions Should(this CommandResult commandResult) => new CommandResultAssertions(commandResult);
```
Comment (MiYanni) [inline test/Microsoft.NET.TestFramework/Assertions/CommandResultExtensions.cs:10]: This file seems to have taken the Installer repo's history. Though, this change is fine for the file.

File: test/EndToEnd.Tests/GivenDotNetLinuxInstallers.cs
Comment (MiYanni) [inline test/EndToEnd.Tests/GivenDotNetLinuxInstallers.cs:1]: This file also seems to have taken the Installer repo's history. Though, this change is fine for the file.

**Review by nagilson — APPROVED** [review]

**Review by Forgind — APPROVED** [review]

**Pull request merged**